operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -659,12 +703,11 @@ private void internalMakeRightDomainPresence() {
               Component.createFor(liveInfo, delegate.getVersion(),
                   PodAwaiterStepFactory.class, delegate.getPodAwaiterStepFactory(getNamespace()),
                   V1SubjectRulesReviewStatus.class, delegate.getSubjectRulesReviewStatus(getNamespace())));
-
       runDomainPlan(
             getDomain(),
             getDomainUid(),
             getNamespace(),
-            new StepAndPacket(createSteps(), packet),
+            new StepAndPacket(createPopulatePacketServerMapsStep(createSteps()), packet),
```

Comment:
Wouldn't it make sense to add the new step into `createSteps`, in the branch where we aren't shutting down the domain?

Reply:
I thought about that, but noticed that createSteps is called in other places outside of the make-right code path.

Reply:
Only in the case where createSteps is called from the make-right code path (internalMakeRightDomainPresence method), the packet does not already have the server maps;

Reply:
Then pull that whole line, 
```
new StepAndPacket(createPopulatePacketServerMapsStep(createSteps()), packet)
```

into its own method, such as `createDomainPlanSteps`. There is otherwise too much going on in that code block.